js/homemenu.js

```diff
@@ -1,12 +1,17 @@
-(function ($, homeMenuItem) {
-  var $homeMenu, $homeMenuLink;
+(function ($, _, homeMenuItem) {
+  var $homeMenu, $homeMenuLink, isHomeMenuReplaced;
 
-  $(document).ready(function () {
-    $homeMenu = $('#civicrm-menu [data-name="Home"]');
-    $homeMenuLink = $homeMenu.find('> a');
-
-    replaceHomeMenu();
-  });
+  /**
+   * We run the home menu replacing both on document ready and when the
+   * CiviCRM Menu has loaded because CiviCRM appends the menu at different times
+   * and we can't determine when it was appended.
+   *
+   * See: https://github.com/civicrm/civicrm-core/blob/master/js/crm.menubar.js#L28-L44
+   */
+  (function init () {
+    $(document).ready(runHomeMenuReplacing);
```

Comment:
Right now, `runHomeMenuReplacing` is checking if home menu should be replaced, and also replacing the menu, we should divide this into two separate functions.

```javascript
$(document).ready(function () {
 if (checkIfHomeMenuShouldBeReplaced()) {
    replaceHomeMenu();
 }
});
```